src/biogeophys/WaterType.F90

```diff
@@ -300,14 +301,22 @@ subroutine DoInit(this, bounds, &
     real(r8)         , intent(in) :: watsat_col(bounds%begc:, 1:)            ! volumetric soil water at saturation (porosity)
     real(r8)         , intent(in) :: t_soisno_col(bounds%begc:, -nlevsno+1:) ! col soil temperature (Kelvin)
     logical          , intent(in) :: use_aquifer_layer ! whether an aquifer layer is used in this run
+    character(len=*) , intent(in) , optional    :: NLFilename ! Namelist filename
```

Comment:
Make this a required argument, it'll simplify the code a bit.

Reply:
To do this the call to DoInit in the unit-test subroutine, just needs to send NLFilename=" " and pass that down.

Reply:
Done